kernel/kheap.c

```diff
@@ -116,6 +116,21 @@ void *k_heap_alloc(struct k_heap *heap, size_t bytes, k_timeout_t timeout)
 	return ret;
 }
 
+void *k_heap_realloc(struct k_heap *heap, void *ptr, size_t bytes, k_timeout_t timeout)
```

Comment:
`timeout` should be used.
Maybe you can refer `k_heap_aligned_alloc`

Reply:
I changed it accordingly.